scripts/circleci/release/installScalyrAgentV2.sh

```diff
@@ -405,24 +434,49 @@ else
-  run_command "apt-get -y install gnupg"
-  # Now install the repo package, which must be done using `dpkg` since it is a package on local disk.
-  # Do not use `apt-get install` here. Even though it does now support installing a package from local disk, older
-  # versions run by our customers do not.
-  run_command "dpkg -i $TMPDIR/*.deb";
-
-  # We need an apt-get update to force apt to refresh the list of repositories
-  # and available packages.  Since this will cause all to refresh,
-  # it takes a little bit of time, but we do not have a choice.
-  # Also, we specifically check for an error some distros experience where apt-transport-https is not installed to
-  # give a better error message.  We cannot depend on that package because not all separate out the https on its own.
-  lecho "  Refreshing available package list (may take several minutes)";
-  run_command "apt-get update" check_for_https_error;
-
-  lecho "  Updating Scalyr yum repository configuration files using repository.";
-  run_command "apt-get -y install scalyr-repo";
+  if $USE_BOOTSTRAP_PACKAGES; then
+    echo "  Installing Scalyr apt repository configuration files.";
+    # Manually install dependencies for the repo package, since `dpkg -i` doesn't install them for you.
+    run_command "apt-get -y install gnupg"
+    # Now install the repo package, which must be done using `dpkg` since it is a package on local disk.
+    # Do not use `apt-get install` here. Even though it does now support installing a package from local disk, older
+    # versions run by our customers do not.
+    run_command "dpkg -i $TMPDIR/*.deb";
+
+    # We need an apt-get update to force apt to refresh the list of repositories
+    # and available packages.  Since this will cause all to refresh,
+    # it takes a little bit of time, but we do not have a choice.
+    # Also, we specifically check for an error some distros experience where apt-transport-https is not installed to
+    # give a better error message.  We cannot depend on that package because not all separate out the https on its own.
+    lecho "  Refreshing available package list (may take several minutes)";
+    run_command "apt-get update" check_for_https_error;
+
+    lecho "  Updating Scalyr yum repository configuration files using repository.";
+    run_command "apt-get -y install scalyr-repo";
+  else
+    # update repository and install tools that may be needed.
+    run_command "apt-get -y update";
+
+    echo "Install required dependencies."
+    export DEBIAN_FRONTEND=noninteractive
+    run_command "apt-get install -y gnupg"
+
+
+    # initialize gpg in case if it has been freshly installed.
+    gpg --update-trustdb
+
+    echo "Adding the public key."
+    run_command "gpg --no-default-keyring --keyring gnupg-ring:/etc/apt/trusted.gpg.d/scalyr.gpg --keyserver ${KEYSERVER_URL} --recv ${PUBLIC_KEY_FINGERPRINT}"
+
+    # change permissions for the gpg key.
+    chmod 660 /etc/apt/trusted.gpg.d/scalyr.gpg
```

Comment:
This is a public key so ``664`` or ``644`` should be fine (aka it can also be readable by others).

In fact, it seems that's what other software does:

```bash
$ ls -la /etc/apt/trusted.gpg.d
total 28
drwxr-xr-x 2 root root 4096 Dec 11 21:06 .
drwxr-xr-x 7 root root 4096 Dec 11 21:06 ..
-rw-r--r-- 1 root root 2796 Apr  9  2020 ubuntu-keyring-2012-archive.gpg
-rw-r--r-- 1 root root 2794 Apr  9  2020 ubuntu-keyring-2012-cdimage.gpg
-rw-r--r-- 1 root root 1733 Apr  9  2020 ubuntu-keyring-2018-archive.gpg
```

Reply:
EDIT: Sorry, looks like I suggested this in the past - https://github.com/scalyr/scalyr-agent-2/pull/679#discussion_r549301213

I imagine not readable by others should also work, but since it's a public key, making it readable (but not writable) by others should be fine as well.

Reply:
Yeah, changed.